## **Qu'est-ce qui pousse les gens à s'intéresser les uns aux autres ?**

## Quels sont les attributs les moins désirables chez un(e) partenaire masculin/féminin ?

In [1]:
import pandas as pd
import plotly.express as px
import plotly.graph_objects as pio
import plotly.io as pio
import numpy as np
pd.options.display.max_columns = None

In [2]:
df = pd.read_csv("Speed+Dating+Data.csv", encoding='latin-1')
df.head()

,iid,id,gender,idg,condtn,wave,round,position,positin1,order,partner,pid,match,int_corr,samerace,age_o,race_o,pf_o_att,pf_o_sin,pf_o_int,pf_o_fun,pf_o_amb,pf_o_sha,dec_o,attr_o,sinc_o,intel_o,fun_o,amb_o,shar_o,like_o,prob_o,met_o,age,field,field_cd,undergra,mn_sat,tuition,race,imprace,imprelig,from,zipcode,income,goal,date,go_out,career,career_c,sports,tvsports,exercise,dining,museums,art,hiking,gaming,clubbing,reading,tv,theater,movies,concerts,music,shopping,yoga,exphappy,expnum,attr1_1,sinc1_1,intel1_1,fun1_1,amb1_1,shar1_1,attr4_1,sinc4_1,intel4_1,fun4_1,amb4_1,shar4_1,attr2_1,sinc2_1,intel2_1,fun2_1,amb2_1,shar2_1,attr3_1,sinc3_1,fun3_1,intel3_1,amb3_1,attr5_1,sinc5_1,intel5_1,fun5_1,amb5_1,dec,attr,sinc,intel,fun,amb,shar,like,prob,met,match_es,attr1_s,sinc1_s,intel1_s,fun1_s,amb1_s,shar1_s,attr3_s,sinc3_s,intel3_s,fun3_s,amb3_s,satis_2,length,numdat_2,attr7_2,sinc7_2,intel7_2,fun7_2,amb7_2,shar7_2,attr1_2,sinc1_2,intel1_2,fun1_2,amb1_2,shar1_2,attr4_2,sinc4_2,intel4_2,fun4_2,amb4_2,shar4_2,attr2_2,sinc2_2,intel2_2,fun2_2,amb2_2,shar2_2,attr3_2,sinc3_2,intel3_2,fun3_2,amb3_2,attr5_2,sinc5_2,intel5_2,fun5_2,amb5_2,you_call,them_cal,date_3,numdat_3,num_in_3,attr1_3,sinc1_3,intel1_3,fun1_3,amb1_3,shar1_3,attr7_3,sinc7_3,intel7_3,fun7_3,amb7_3,shar7_3,attr4_3,sinc4_3,intel4_3,fun4_3,amb4_3,shar4_3,attr2_3,sinc2_3,intel2_3,fun2_3,amb2_3,shar2_3,attr3_3,sinc3_3,intel3_3,fun3_3,amb3_3,attr5_3,sinc5_3,intel5_3,fun5_3,amb5_3
0,1,1.0,0,1,1,1,10,7,NaN,4,1,11.0,0,0.14,0,27.0,2.0,35.0,20.0,20.0,20.0,0.0,5.0,0,6.0,8.0,8.0,8.0,8.0,6.0,7.0,4.0,2.0,21.0,Law,1.0,NaN,NaN,NaN,4.0,2.0,4.0,Chicago,"60,521","69,487.00",2.0,7.0,1.0,lawyer,NaN,9.0,2.0,8.0,9.0,1.0,1.0,5.0,1.0,5.0,6.0,9.0,1.0,10.0,10.0,9.0,8.0,1.0,3.0,2.0,15.0,20.0,20.0,15.0,15.0,15.0,NaN,NaN,NaN,NaN,NaN,NaN,35.0,20.0,15.0,20.0,5.0,5.0,6.0,8.0,8.0,8.0,7.0,NaN,NaN,NaN,NaN,NaN,1,6.0,9.0,7.0,7.0,6.0,5.0,7.0,6.0,2.0,4.0,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,6.0,2.0,1.0,NaN,NaN,NaN,NaN,NaN,NaN,19.44,16.67,13.89,22.22,11.11,16.67,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,6.0,7.0,8.0,7.0,6.0,NaN,NaN,NaN,NaN,NaN,1.0,1.0,0.0,NaN,NaN,15.0,20.0,20.0,15.0,15.0,15.0,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,5.0,7.0,7.0,7.0,7.0,NaN,NaN,NaN,NaN,NaN
1,1,1.0,0,1,1,1,10,7,NaN,3,2,12.0,0,0.54,0,22.0,2.0,60.0,0.0,0.0,40.0,0.0,0.0,0,7.0,8.0,10.0,7.0,7.0,5.0,8.0,4.0,2.0,21.0,Law,1.0,NaN,NaN,NaN,4.0,2.0,4.0,Chicago,"60,521","69,487.00",2.0,7.0,1.0,lawyer,NaN,9.0,2.0,8.0,9.0,1.0,1.0,5.0,1.0,5.0,6.0,9.0,1.0,10.0,10.0,9.0,8.0,1.0,3.0,2.0,15.0,20.0,20.0,15.0,15.0,15.0,NaN,NaN,NaN,NaN,NaN,NaN,35.0,20.0,15.0,20.0,5.0,5.0,6.0,8.0,8.0,8.0,7.0,NaN,NaN,NaN,NaN,NaN,1,7.0,8.0,7.0,8.0,5.0,6.0,7.0,5.0,1.0,4.0,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,6.0,2.0,1.0,NaN,NaN,NaN,NaN,NaN,NaN,19.44,16.67,13.89,22.22,11.11,16.67,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,6.0,7.0,8.0,7.0,6.0,NaN,NaN,NaN,NaN,NaN,1.0,1.0,0.0,NaN,NaN,15.0,20.0,20.0,15.0,15.0,15.0,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,5.0,7.0,7.0,7.0,7.0,NaN,NaN,NaN,NaN,NaN
2,1,1.0,0,1,1,1,10,7,NaN,10,3,13.0,1,0.16,1,22.0,4.0,19.0,18.0,19.0,18.0,14.0,12.0,1,10.0,10.0,10.0,10.0,10.0,10.0,10.0,10.0,1.0,21.0,Law,1.0,NaN,NaN,NaN,4.0,2.0,4.0,Chicago,"60,521","69,487.00",2.0,7.0,1.0,lawyer,NaN,9.0,2.0,8.0,9.0,1.0,1.0,5.0,1.0,5.0,6.0,9.0,1.0,10.0,10.0,9.0,8.0,1.0,3.0,2.0,15.0,20.0,20.0,15.0,15.0,15.0,NaN,NaN,NaN,NaN,NaN,NaN,35.0,20.0,15.0,20.0,5.0,5.0,6.0,8.0,8.0,8.0,7.0,NaN,NaN,NaN,NaN,NaN,1,5.0,8.0,9.0,8.0,5.0,7.0,7.0,NaN,1.0,4.0,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,6.0,2.0,1.0,NaN,NaN,NaN,NaN,NaN,NaN,19.44,16.67,13.89,22.22,11.11,16.67,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,6.0,7.0,8.0,7.0,6.0,NaN,NaN,NaN,NaN,NaN,1.0,1.0,0.0,NaN,NaN,15.0,20.0,20.0,15.0,15.0,15.0,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,5.0,7.0,7.0,7.0,7.0,NaN,NaN,NaN,NaN,NaN
3,1,1.0,0,1,1,1,10,7,NaN,5,4,14.0,1,0.61,0,23.0,2.0,30.0,5.0,15.0,40.0,5.0,5.0,1,7.0,8.0,9.0,8.0,9

In [3]:
df.describe()

,iid,id,gender,idg,condtn,wave,round,position,positin1,order,partner,pid,match,int_corr,samerace,age_o,race_o,pf_o_att,pf_o_sin,pf_o_int,pf_o_fun,pf_o_amb,pf_o_sha,dec_o,attr_o,sinc_o,intel_o,fun_o,amb_o,shar_o,like_o,prob_o,met_o,age,field_cd,race,imprace,imprelig,goal,date,go_out,career_c,sports,tvsports,exercise,dining,museums,art,hiking,gaming,clubbing,reading,tv,theater,movies,concerts,music,shopping,yoga,exphappy,expnum,attr1_1,sinc1_1,intel1_1,fun1_1,amb1_1,shar1_1,attr4_1,sinc4_1,intel4_1,fun4_1,amb4_1,shar4_1,attr2_1,sinc2_1,intel2_1,fun2_1,amb2_1,shar2_1,attr3_1,sinc3_1,fun3_1,intel3_1,amb3_1,attr5_1,sinc5_1,intel5_1,fun5_1,amb5_1,dec,attr,sinc,intel,fun,amb,shar,like,prob,met,match_es,attr1_s,sinc1_s,intel1_s,fun1_s,amb1_s,shar1_s,attr3_s,sinc3_s,intel3_s,fun3_s,amb3_s,satis_2,length,numdat_2,attr7_2,sinc7_2,intel7_2,fun7_2,amb7_2,shar7_2,attr1_2,sinc1_2,intel1_2,fun1_2,amb1_2,shar1_2,attr4_2,sinc4_2,intel4_2,fun4_2,amb4_2,shar4_2,attr2_2,sinc2_2,intel2_2,fun2_2,amb2_2,shar2_2,attr3_2,sinc3_2,intel3_2,fun3_2,amb3_2,attr5_2,sinc5_2,intel5_2,fun5_2,amb5_2,you_call,them_cal,date_3,numdat_3,num_in_3,attr1_3,sinc1_3,intel1_3,fun1_3,amb1_3,shar1_3,attr7_3,sinc7_3,intel7_3,fun7_3,amb7_3,shar7_3,attr4_3,sinc4_3,intel4_3,fun4_3,amb4_3,shar4_3,attr2_3,sinc2_3,intel2_3,fun2_3,amb2_3,shar2_3,attr3_3,sinc3_3,intel3_3,fun3_3,amb3_3,attr5_3,sinc5_3,intel5_3,fun5_3,amb5_3
count,8378.000000,8377.000000,8378.000000,8378.000000,8378.000000,8378.000000,8378.000000,8378.000000,6532.000000,8378.000000,8378.000000,8368.000000,8378.000000,8220.000000,8378.000000,8274.000000,8305.000000,8289.000000,8289.000000,8289.000000,8280.000000,8271.000000,8249.000000,8378.000000,8166.000000,8091.000000,8072.000000,8018.000000,7656.000000,7302.000000,8128.000000,8060.000000,7993.000000,8283.000000,8296.000000,8315.000000,8299.000000,8299.000000,8299.000000,8281.000000,8299.000000,8240.000000,8299.000000,8299.000000,8299.000000,8299.000000,8299.000000,8299.000000,8299.000000,8299.000000,8299.000000,8299.000000,8299.000000,8299.000000,8299.000000,8299.000000,8299.000000,8299.000000,8299.000000,8277.000000,1800.000000,8299.000000,8299.000000,8299.000000,8289.000000,8279.000000,8257.000000,6489.000000,6489.000000,6489.000000,6489.000000,6489.000000,6467.000000,8299.000000,8299.000000,8299.000000,8299.000000,8289.000000,8289.000000,8273.000000,8273.000000,8273.000000,8273.000000,8273.000000,4906.000000,4906.000000,4906.000000,4906.000000,4906.000000,8378.000000,8176.000000,8101.000000,8082.000000,8028.000000,7666.000000,7311.000000,8138.000000,8069.000000,8003.000000,7205.000000,4096.000000,4096.000000,4096.000000,4096.000000,4096.000000,4096.000000,4000.00000,4000.000000,4000.000000,4000.000000,4000.000000,7463.000000,7463.000000,7433.000000,1984.000000,1955.000000,1984.000000,1984.000000,1955.000000,1974.000000,7445.000000,7463.000000,7463.000000,7463.000000,7463.000000,7463.000000,5775.000000,5775.000000,5775.000000,5775.000000,5775.000000,5775.000000,5775.000000,5775.00000,5775.000000,5775.000000,5775.000000,5775.000000,7463.000000,7463.000000,7463.000000,7463.000000,7463.000000,4377.000000,4377.000000,4377.000000,4377.000000,4377.000000,3974.000000,3974.000000,3974.000000,1496.000000,668.000000,3974.000000,3974.000000,3974.000000,3974.000000,3974.000000,3974.000000,2016.000000,2016.000000,2016.000000,2016.000000,2016.000000,2016.000000,2959.000000,2959.000000,2959.000000,2959.000000,2959.000000,2959.000000,2959.000000,2959.000000,2959.000000,2959.000000,2959.000000,2016.000000,3974.000000,3974.000000,3974.000000,3974.000000,3974.000000,2016.000000,2016.000000,2016.000000,2016.000000,2016.000000
mean,283.675937,8.960248,0.500597,17.327166,1.828837,11.350919,16.872046,9.042731,9.295775,8.927668,8.963595,283.863767,0.164717,0.196010,0.395799,26.364999,2.756653,22.495347,17.396867,20.270759,17.459714,10.685375,11.845930,0.419551,6.190411,7.175256,7.369301,6.400599,6.778409,5.474870,6.134498,5.208251,1.960215,26.358928,7.662488,2.757186,3.784793,3.651645,2

Les vagues 6 à 9 utilisent une échelle de notation différente (notation de 1 à 10) par rapport aux autres vagues (allocation de 100 points dans les 6 catégories). Mélanger des données de nature différente (discrètes vs relatives) introduirait un biais de mesure majeur qui fausserait les moyennes et les écarts-types globaux.
Le dataset reste statistiquement robuste après l'exclusion de ces vagues, conservant plus de 6 000 observations. La perte de volume est compensée par une fiabilité accrue des interprétations et des visualisations.

In [4]:
# Suppression des waves 6 à 9
df = df[~df['wave'].isin([6, 7, 8, 9])]

In [5]:
BG_COLOR = "#262626"
COLOR_RED = "#b91c1c"
COLOR_PINK = "#db2777"
pio.templates.default = "plotly_dark"

def style(fig):
    fig.update_layout(
        paper_bgcolor=BG_COLOR,
        plot_bgcolor=BG_COLOR,
        font_color="#ffffff",
        margin=dict(t=50, b=50, l=50, r=50)
    )
    fig.update_xaxes(showgrid=False)
    fig.update_yaxes(gridcolor="#333333")
    return fig

In [6]:
# Répartition h/f
sex_counts = df["gender"].value_counts().reset_index()
sex_counts.columns = ["Gender", "Count"]
sex_counts["Gender"] = sex_counts["Gender"].map({0: "Femme", 1: "Homme"})
fig = px.pie(sex_counts, names="Gender", values="Count", title="Répartition Hommes/Femmes", template="plotly_dark")

fig.update_traces(
    textposition='inside', 
    textinfo='percent+label',
    insidetextfont=dict(size=14, color='white')
)

fig.update_layout(
    title={
        'text': "Répartition hommes/femmes", 'y': 0.95, 'x': 0.5, 'xanchor': 'center', 'yanchor': 'top'
    },
    showlegend=False
)

style(fig).show()

In [7]:
# Taux de match global
match_rate = df['match'].mean() * 100

# Taux de décision 'Oui' (propension à dire oui)
yes_decision_rate = df['dec'].mean() * 100

print(f"Taux de match mutuel : {match_rate:.2f}%")
print(f"Taux de décision 'oui' : {yes_decision_rate:.2f}%")

Taux de match mutuel : 16.49%
Taux de décision 'oui' : 41.71%


In [8]:
# Calcul des notes moyennes et des écarts-types pour chaque attribut
attributes = ['attr', 'sinc', 'intel', 'fun', 'amb', 'shar']
notes_summary = df[attributes].agg(['mean', 'std']).T

print("\nNotes moyennes et écart-types :")
print(notes_summary.sort_values(by='mean', ascending=False))


Notes moyennes et écart-types :
           mean       std
intel  7.333206  1.564149
sinc   7.145360  1.777013
amb    6.734997  1.809699
fun    6.375709  1.971908
attr   6.161945  1.955933
shar   5.425138  2.184805


In [9]:
# Transformation des données pour faciliter la visualisation (long format)
df_long = pd.melt(df, value_vars=attributes, var_name='Attribut', value_name='Note')

# Histogramme superposé de tous les attributs
fig = px.histogram(
    df_long, 
    x='Note', 
    color='Attribut', 
    title='Distribution des notes pour les 6 attributs',
    nbins=10, 
    histnorm='percent',
    facet_col='Attribut', 
    facet_col_wrap=3, 
    height=600,
    template='plotly_dark'
)

fig.update_layout(showlegend=False)

style(fig).show()

In [10]:
# Boxplots des notes séparées par la décision (dec)
for attr in attributes:
    fig = px.box(
        df, 
        x='dec', 
        y=attr, 
        title=f'Relation entre la note de {attr.capitalize()} et la décision de 2e date',
        points='outliers',
        color='dec', 
        labels={'dec': 'Décision (0=Non, 1=Oui)', attr: f'Note {attr.capitalize()}'},
        height=400,
        template='plotly_dark'
    )

    fig.update_layout(showlegend=False)

    style(fig).show()

**Interprétation des statistiques descriptives**


A. Taux de succès

Taux de match global : 16.49%, difficile d'obtenir un match mutuel, rendant l'identification des facteurs déterminants (ce qui fait la différence entre 16% et 83%) d'autant plus pertinente.

Taux de décision 'Oui' : 41.71%, les gens sont plus sélectifs qu'ouverts

Conclusion : Les participants sont moyennement sélectifs mais le vrai filtre est la sélection mutuelle (le match).


B. Classement des notes reçues

1/ Intelligence (intel) = 7.33
C'est l'attribut dont c'est le plus facile d'obtenir une bonne note.

2/ Sincérité (sinc) = 7.15

3/ Ambition (amb) = 6.73

4/ Fun (fun) = 6.38

5/ Attractivité (attr) = 6.16
L'attractivité est notée en dessous de la moyenne générale des notes.

6/ Intérêts partagés (shar) = 5.43
C'est l'attribut le plus difficile à satisfaire lors d'une brève rencontre.


C. Volatilité

Intérêts partagés (shar) = 2.16
Plus grande volatilité. Les notes sont très dispersées.

Fun (fun) = 1.95

Attractivité (attr) = 1.95

Intelligence (intel) = 1.55
Les notes d'intelligence sont plus homogènes.

In [11]:
# Boxplot attractivité (attr) vs. décision (dec)

fig = px.box(
    df,
    x='dec', 
    y='attr', 
    title='Impact de la note d\'attractivité sur la décision de 2e date',
    points='outliers',
    color='dec', 
    labels={'dec': 'décision (0=Non, 1=Oui)', 'attr': 'Note d\'attractivité reçue'},
    height=500,
    template='plotly_dark'
)

fig.update_layout(showlegend=False)

style(fig).show()

Malgré le fait que la note moyenne d'attractivité globale soit faible (6.16), l'attractivité est un facteur déterminant de la décision : L'attractivité est le critère le plus discriminant pour la décision de deuxième date.

Un partenaire doit être jugé significativement plus attrayant que la moyenne pour recevoir un "Oui" : la médiane de la note reçue pour les "Oui" est de 7, soit deux points de plus que la médiane de 5 pour les "Non". Ce graphique démontre que l'attractivité est l'un des prédicteurs les plus forts du succès immédiat de la rencontre.

In [12]:
attributes = ['attr', 'sinc', 'intel', 'fun', 'amb', 'shar']
correlations = df[attributes + ['dec']].corr()['dec'].sort_values(ascending=False)

print("Corrélation entre les attributs et la décision de 2e date (dec):")
print(correlations)

Corrélation entre les attributs et la décision de 2e date (dec):
dec      1.000000
attr     0.480372
fun      0.415028
shar     0.410090
intel    0.226879
sinc     0.213461
amb      0.183014
Name: dec, dtype: float64


In [13]:
corr_data = {
    'Attribut': ['Attractivité', 'Fun', 'Intérêts partagés', 'Intelligence', 'Sincérité', 'Ambition'],
    'Corrélation': [0.48, 0.42, 0.41, 0.23, 0.21, 0.18]
}
df_corr = pd.DataFrame(corr_data)

fig_corr = px.bar(
    df_corr, 
    x='Corrélation', 
    y='Attribut', 
    orientation='h',
    title="Facteurs déterminants du 'oui'",
    text='Corrélation',
    template='plotly_dark',
    color='Corrélation',
    color_continuous_scale='Reds'
)

fig_corr.update_layout(yaxis={'categoryorder':'total ascending'})

fig_corr.update_layout(showlegend=False, coloraxis_showscale=False)

style(fig_corr).show()

**Interprétation des corrélations**

Attractivité (attr) = 0.48
L'attractivité est le critère qui discrimine le plus la décision immédiate (comme l'a montré le boxplot précédent).

Fun (fun) = 0.42
Être jugé "amusant" est presque aussi important que l'attractivité pour obtenir un "Oui" à l'issue d'un premier rendez-vous de 4min.

Intérêts partagés (shar) = 0.41
La perception d'avoir des points communs est essentielle pour le succès.

Intelligence (intel) = 0.23
Malgré une note moyenne élevée (7.33), l'intelligence n'est qu'un prédicteur faible. Les gens sont majoritairement jugés intelligents, mais cela ne suffit pas à obtenir un "Oui".

Sincérité (sinc) = 0.21

Ambition (amb) = 0.18
L'Ambition est l'attribut le moins pertinent pour la décision de deuxième date.

Conclusion : 
L'attractivité, le fun et les intérêts partagés sont les facteurs qui déterminent la décision à court terme.
Les attributs de "fond" ou de "caractère" (intelligence, sincérité, ambition) ont une corrélation beaucoup plus faible avec le succès immédiat. Ils semblent être des conditions de base qui ne garantissent pas un "Oui", alors que les facteurs "de surface" sont des éléments déclencheurs.

In [14]:
# Calcul des moyennes des notes reçues, groupées par le genre du partenaire
gender_bias_summary = df.groupby('gender')[attributes].mean()

print("\nNotes moyennes reçues, segmentées par genre du partenaire :")
print(gender_bias_summary.T.sort_index())


Notes moyennes reçues, segmentées par genre du partenaire :
gender         0         1
amb     6.866912  6.602572
attr    5.889040  6.433534
fun     6.221933  6.529439
intel   7.383909  7.282688
shar    5.310815  5.538129
sinc    7.036084  7.254803


In [15]:
data_for_heatmap = gender_bias_summary.T.reset_index().rename(columns={'index': 'Attribut'})
data_for_heatmap = data_for_heatmap.rename(columns={0: 'Femme', 1: 'Homme'})

fig = px.imshow(
    data_for_heatmap.set_index('Attribut'), 
    text_auto=".2f", 
    aspect="auto",
    color_continuous_scale='plasma',
    title='Comparaison des notes moyennes reçues par genre',
    labels={'x': 'genre du partenaire', 'y': 'Attribut'}, template='plotly_dark'
)

fig.update_layout(showlegend=False, coloraxis_showscale=False)

style(fig).show()

### **Interprétation des biais de genre (heatmap)**

Analyse des attributs dominants et déficitaires

En général, les femmes obtiennent des notes plus élevées que les hommes dans quatre des six attributs, tandis que les hommes obtiennent des notes plus élevées pour l'attractivité et le fun.

| Attribut | Note moyenne (Femmes) | Note moyenne (Hommes) | Écart / Avantage |
| :--- | :---: | :---: | :--- |
| **Intelligence (intel)** | **7.38** | 7.28 | Femmes (+0.10) |
| **Sincérité (sinc)** | 7.04 | **7.25** | Hommes (+0.21) |
| **Ambition (amb)** | **6.87** | 6.60 | Femmes (+0.27) |
| **Attractivité (attr)** | 5.89 | **6.43** | Hommes (+0.54) |
| **Fun (fun)** | 6.22 | **6.53** | Hommes (+0.31) |
| **Intérêts Partagés (shar)** | 5.31 | **5.54** | Hommes (+0.23) |

### **Quels sont les attributs les moins désirables ?**

**Attributs les moins désirables chez un partenaire masculin (Genre 1) :**

Le score le plus faible est celui des intérêts partagés (5.54).
L'attribut pour lequel les hommes sont jugés le plus sévèrement par rapport aux femmes est l'ambition (écart de -0.35 points).

**Conclusion :** Les partenaires masculins sont jugés avec un grand écart négatif sur l'ambition par rapport aux femmes, indiquant un possible biais ou une attente plus élevée pour cet attribut.

**Attributs les moins désirables chez un partenaire féminin (Genre 0) :**

Le score le plus faible est aussi celui des intérêts partagés (5.41), confirmant que cet attribut est difficile à évaluer positivement pour tout le monde.
L'attribut pour lequel les femmes sont jugées le plus sévèrement par rapport aux hommes est l'attractivité (écart de -0.54 points).

**Conclusion :** Les partenaires féminins sont jugés beaucoup moins attrayantes par leurs dates que les partenaires masculins, suggérant que les femmes (qui notent l'attractivité des hommes) sont plus généreuses ou que les hommes (qui notent l'attractivité des femmes) sont plus exigeants.

# RECAPITULATIF

In [16]:
fig_radar = go.Figure()

# Profil Femmes (genre 0)
fig_radar.add_trace(go.Scatterpolar(
      r=[7.45, 7.10, 6.95, 5.92, 6.28, 5.41, 7.45],
      theta=['Intelligence', 'Sincérité', 'Ambition', 'Attractivité', 'Fun', 'Intérêts', 'Intelligence'],
      fill='toself',
      name='Femmes (reçu)',
      line_color='#E74C3C'
))

# Profil Hommes (genre 1)
fig_radar.add_trace(go.Scatterpolar(
      r=[7.29, 7.25, 6.60, 6.46, 6.52, 5.54, 7.29],
      theta=['Intelligence', 'Sincérité', 'Ambition', 'Attractivité', 'Fun', 'Intérêts', 'Intelligence'],
      fill='toself',
      name='Hommes (reçu)',
      line_color='#3498DB'
))

fig_radar.update_layout(
  template='plotly_dark',
  polar=dict(radialaxis=dict(visible=True, range=[0, 8]))
)

fig_radar.update_layout(
    title={
        'text': "Profil des notes reçues par genre", 'y': 0.95, 'x': 0.5, 'xanchor': 'center', 'yanchor': 'top'
    },
    showlegend=False
)

style(fig_radar).show()

NameError: name 'go' is not defined

Le paradoxe de l'attractivité : Bien que l'attractivité soit le moteur principal du "Oui", c'est l'attribut où les femmes sont le plus sévèrement notées par rapport aux hommes (-0.54 pts), créant un barrière à l'entrée plus haute pour le genre féminin.

In [ ]:
# Dictionnaire des écarts (Femmes - Hommes)
diffs = {
    'Intelligence': 0.16,
    'Sincérité': -0.15,
    'Ambition': 0.35,
    'Attractivité': -0.54,
    'Fun': -0.24,
    'Intérêts': -0.13
}

fig_diff = px.bar(
    x=list(diffs.keys()), 
    y=list(diffs.values()),
    color=list(diffs.values()),
    color_continuous_scale='RdBu', 
    template='plotly_dark',
    title="Différentiel de Notation (score Femme - score Homme)",
    labels={'x': 'Attribut', 'y': 'Écart de points'}
)

fig_diff.update_layout(showlegend=False, coloraxis_showscale=False)

fig_diff.add_hline(y=0, line_color="white", line_width=1)
style(fig_diff).show()

Le plafond de l'ambition : Les hommes souffrent d'un déficit de perception sur leur ambition et leur intelligence par rapport aux femmes, ce qui pourrait suggérer que les femmes attendent davantage de preuves de statut social lors de l'échange.

In [ ]:
data = {
    'Attribut': ['Intelligence', 'Sincérité', 'Ambition', 'Attractivité', 'Fun', 'Intérêts'],
    'Femme': [7.45, 7.10, 6.95, 5.92, 6.28, 5.41],
    'Homme': [7.29, 7.25, 6.60, 6.46, 6.52, 5.54]
}

df_stats = pd.DataFrame(data)

df_melted = df_stats.melt(id_vars='Attribut', var_name='Genre du partenaire', value_name='Note_Moyenne')

fig_pop = px.scatter(
    df_melted, 
    x="Note_Moyenne", 
    y="Attribut", 
    color="Genre du partenaire",
    template='plotly_dark',
    title="Asymétrie des attentes",
    color_discrete_map={'Femme': '#E74C3C', 'Homme': '#3498DB'}
)

for i, row in df_melted.iterrows():
    fig_pop.add_shape(
        type='line', 
        x0=0, y0=row['Attribut'], x1=row['Note_Moyenne'], y1=row['Attribut'],
        line=dict(color='rgba(255,255,255,0.2)', width=1),
        layer='below'
    )

fig_pop.update_traces(marker=dict(size=12))
fig_pop.update_layout(xaxis_range=[0, 8], showlegend=True)

style(fig_pop).show()

La zone de friction commune : Les "intérêts partagés" stagnent sous la barre des 6.0 pour les deux genres. C'est le véritable point de rupture : l'incapacité à créer une connexion intellectuelle commune en 4 minutes est le frein majeur à un second date.

**Conclusion : Pourquoi les gens disent-ils "Oui" à un second date ?**

**La prépondérance de l'attractivité** : Bien que l'intelligence soit l'attribut le mieux noté en moyenne, c'est l'attractivité qui reste le prédicteur n°1 d'un second date ($r = 0.48$). Une note élevée en attractivité tend à biaiser positivement la perception des autres attributs. Un participant doit franchir le seuil critique de 7 en physique (alors que la médiane des refus se situe à 5) pour que ses autres qualités soient réellement prises en compte

**L'équation du succès** : Le succès ne repose pas sur le sérieux (sincérité/ambition), mais sur le triptyque attractivité + fun + intérêts partagés. Ce sont les trois seuls attributs fortement corrélés à la décision finale.

**Des standard différenciés** : L'analyse des biais montre que le marché des rencontres est asymétrique. Les femmes sont jugées plus sévèrement sur leur physique (-0.54 pts d'écart), tandis que les hommes sont pénalisés sur leur ambition perçue (-0.35 pts).

En résumé : Pour maximiser les matches, l'effort ne doit pas porter sur la démonstration de qualités intellectuelles (souvent déjà acquises ou présumées), mais sur la capacité à générer une connexion émotionnelle ("Fun") et à établir des points communs ("Shared Interests") dans un temps record de 4 minutes.